v2/googlecloud-to-googlecloud/src/test/java/com/google/cloud/teleport/v2/utils/SpannerUtilsTest.java

```diff
@@ -0,0 +1,307 @@
+
+    JsonObject expected = new JsonObject();
+    expected.addProperty("value", "test");
+    assertThat(actual).isEqualTo(expected);
+  }
+
+  @Test
+  public void testDateStructToJson() {
+    Struct testStruct =
+        Struct.newBuilder().set("value").to(Value.date(Date.fromYearMonthDay(2020, 3, 15))).build();
+
+    JsonObject actual = convertStructToJson(testStruct);
+
+    JsonObject expected = new JsonObject();
+    expected.addProperty("value", "2020-03-15");
+    assertThat(actual).isEqualTo(expected);
+  }
+
+  @Test
+  public void testTimestampStructToJson() {
+    Struct testStruct =
+        Struct.newBuilder()
+            .set("value")
+            .to(Value.timestamp(Timestamp.ofTimeMicroseconds(10)))
+            .build();
+
+    JsonObject actual = convertStructToJson(testStruct);
+
+    JsonObject expected = new JsonObject();
+    expected.addProperty("value", "1970-01-01T00:00:00.000010000Z");
+    assertThat(actual).isEqualTo(expected);
+  }
+
+  @Test
+  public void nestedStructToJson() {
+    Struct innerStruct = Struct.newBuilder().set("val").to(Value.string("test-string")).build();
+    Struct testStruct = Struct.newBuilder().set("value").to(Value.struct(innerStruct)).build();
+
+    JsonObject actual = convertStructToJson(testStruct);
+
+    JsonObject innerObject = new JsonObject();
+    innerObject.addProperty("val", "test-string");
+    JsonObject expected = new JsonObject();
+    expected.add("value", innerObject);
+    assertThat(actual).isEqualTo(expected);
+  }
+
+  @Test
+  public void testBooleanArrayStructToJson() {
+    Struct testStruct =
+        Struct.newBuilder()
+            .set("value")
+            .to(Value.boolArray(Collections.singletonList(true)))
+            .build();
+
+    JsonObject actual = convertStructToJson(testStruct);
+
+    JsonObject expected = new JsonObject();
+    JsonArray jsonArray = new JsonArray();
+    jsonArray.add(true);
+    expected.add("value", jsonArray);
+    assertThat(actual).isEqualTo(expected);
+  }
+
+  @Test
+  public void testInt64ArrayStructToJson() {
+    Struct testStruct =
+        Struct.newBuilder().set("value").to(Value.int64Array(new long[] {1})).build();
+
+    JsonObject actual = convertStructToJson(testStruct);
+
+    JsonObject expected = new JsonObject();
+    JsonArray jsonArray = new JsonArray();
+    jsonArray.add(1);
+    expected.add("value", jsonArray);
+    assertThat(actual).isEqualTo(expected);
+  }
+
+  @Test
+  public void testFloatArrayStructToJson() {
+    Struct testStruct =
+        Struct.newBuilder().set("value").to(Value.float64Array(new double[] {0.0})).build();
+
+    JsonObject actual = convertStructToJson(testStruct);
+
+    JsonObject expected = new JsonObject();
+    JsonArray jsonArray = new JsonArray();
+    jsonArray.add(0.0);
+    expected.add("value", jsonArray);
+    assertThat(actual).isEqualTo(expected);
+  }
+
+  @Test
+  public void testStringArrayStructToJson() {
+    Struct testStruct =
```

Comment:
Like with the above (and other multi-line structs), this should have `set` and `to` on the same line.

To make the line shorter, you can also static import all the `Value.X` methods.